### PR TITLE
Now omnisharp-sublime work with single files

### PR DESCRIPTION
**Pull request by gordio**
I'm little confused with two exceptions, but can't find better solution.

### CODE REVIEW

File: lib/helpers.py [hunk 1]
```diff
@@ -70,7 +70,14 @@ def current_solution_or_project_json_folder(view):
             solution_file_path = os.path.abspath(solution_file_path)
             return solution_file_path
     else:
-        parentpath = sublime.active_window().folders()[0] #assume parent folder is opened that contains all project folders eg/Web,ClassLib,Tests
+        try:
+            parentpath = sublime.active_window().folders()[0] #assume parent folder is opened that contains all project folders eg/Web,ClassLib,Tests
```
Comment (jchannon) [inline lib/helpers.py]: does this throw? i assume its just empty then we can check for an empty array?

File: lib/helpers.py [hunk 2]
```diff
@@ -70,7 +70,14 @@ def current_solution_or_project_json_folder(view):
             solution_file_path = os.path.abspath(solution_file_path)
             return solution_file_path
     else:
-        parentpath = sublime.active_window().folders()[0] #assume parent folder is opened that contains all project folders eg/Web,ClassLib,Tests
+        try:
+            parentpath = sublime.active_window().folders()[0] #assume parent folder is opened that contains all project folders eg/Web,ClassLib,Tests
+        except Exception:
+            try:
+                parentpath = os.path.dirname(sublime.active_window().active_view().file_name())
+            except Exception:
```
Comment (jchannon) [inline lib/helpers.py]: I wonder if this exception could be prevented if we expand this https://github.com/OmniSharp/omnisharp-sublime/blob/master/lib/helpers.py#L6 to check if the syntax of the file is marked as C#

Comment (gordio) [inline lib/helpers.py]: But possible situation with unsaved C# syntax

File: lib/helpers.py (hunk 1 above)
Comment (gordio) [inline lib/helpers.py]: You right.

File: lib/helpers.py (hunk 2 above)
Comment (jchannon) [inline lib/helpers.py]: If you create a new file that is unsaved and specify a syntax for it in sublime `View - Syntax -C#` we could use that with omnisharp.

**Comment by jchannon** [comment]
Hi, huge apologies that I've not looked at this sooner. Will take a look soon

**Comment by gordio** [comment]
Thank you.

**Comment by jchannon** [comment]
No probs - on the `roslyn` branch we've changed this area of the code so i need to move that to master then add your changes and change anywhere where this code is called :smile:

**Comment by jchannon** [comment]
Hi,

I've cherry picked some quite big changes from the roslyn branch into master. You'll have to do a rebase.

```
git pull upstream master
git rebase master
--sort conflicts
git push myfork master
```

**Comment by jchannon** [comment]
This has been merged but not sure why its not showing here so manually closing the PR.

Thanks for the contribution :+1:

**Pull request closed without merging**